akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -350,6 +350,8 @@ trait FlowOps[+Out, +Mat] {
 
   private final val _identity = (x: Any) ⇒ x
 
+  def recover[T](pf: PartialFunction[Throwable, T]): Repr[T, Mat] = andThen(Recover(pf.asInstanceOf[PartialFunction[Any, Any]]))
```

Comment:
one important (non-obvious) part that must be documented is that the recovery element may overtake other normally emitted elements, since onError is not driven by demand